src/Shape/Shape.js

```diff
@@ -407,14 +411,17 @@ export default class Shape extends BaseClass {
             for (let l = 0; l < labels.length; l++) {
 
               const b = bounds.constructor === Array ? bounds[l] : Object.assign({}, bounds);
+              const r = d.labelConfig && d.labelConfig.rotate ? d.labelConfig.rotate + this._rotate(d, i) : bounds.angle !== undefined ? bounds.angle + this._rotate(d, i) : this._rotate(d, i);
+              const rotationAnchor = this._rotate(d, i) !== 0 && bounds && [bounds.x * -1 || 0, bounds.y * -1 || 0];
```

Comment:
If you're breaking out the above logic, calculate `this._rotate(d, i)` only once, and then use that variable here (trying to cut down on accessor calls, although they seem quick in testing they add up when doing a big viz like a big network).